ush/ufsda/stage.py

```diff
@@ -55,6 +55,111 @@ def gdas_fix(input_fix_dir, working_dir, config):
+    ufsda.r2d2.fetch(r2d2_config)
+    # get gfs metadata
+    r2d2_config['model'] = 'gfs_metadata'
+    r2d2_config['target_file_fmt'] = '{target_dir}/$(valid_date).$(file_type)'
+    r2d2_config['file_type_list'] = ['coupler.res', 'fv_core.res.nc']
+    del r2d2_config['tile']
+    ufsda.r2d2.fetch(r2d2_config)
+
+
+def atm_obs(config):
+    # fetch atm analysis obs
+    r2d2_config = {
+        'start': config['prev_valid_time'],
+        'end': config['prev_valid_time'],
+        'step': config['atm_window_length'],
+        'dump': 'gdas',
+        'experiment': 'oper_gdas',  # change this here and other places to be oper_{dump}
+        'target_dir': config.get('target_dir', config.get('BKG_DIR', './')),
+    }
+    r2d2_config = NiceDict(r2d2_config)
+    # get list of obs to process and their output files
+    obs_list_yaml = config['OBS_LIST']
+    obs_list_config = Configuration(obs_list_yaml)
+    obs_list_config = ufsda.yamltools.iter_config(config, obs_list_config)
+    for ob in obs_list_config['observations']:
+        # first get obs
+        r2d2_config.pop('file_type', None)
+        r2d2_config['type'] = 'ob'
+        r2d2_config['provider'] = 'ncdiag'
+        r2d2_config['start'] = config['window_begin']
+        r2d2_config['end'] = r2d2_config['start']
+        target_file = ob['obs space']['obsdatain']['obsfile']
+        r2d2_config['target_file_fmt'] = target_file
+        r2d2_config['obs_types'] = [ob['obs space']['name']]
+        ufsda.r2d2.fetch(r2d2_config)
+
+
+def bias_obs(config):
+    # fetch bias files
+    r2d2_config = {
+        'start': config['prev_valid_time'],
+        'end': config['prev_valid_time'],
+        'step': config['atm_window_length'],
+        'dump': 'gdas',
+        'experiment': 'oper_gdas',  # change this here and other places to be oper_{dump}
+        'target_dir': config.get('target_dir', config.get('BKG_DIR', './')),
+    }
+    r2d2_config = NiceDict(r2d2_config)
+    # get list of obs to process and their output files
+    obs_list_yaml = config['OBS_LIST']
+    obs_list_config = Configuration(obs_list_yaml)
+    obs_list_config = ufsda.yamltools.iter_config(config, obs_list_config)
+    for ob in obs_list_config['observations']:
+        r2d2_config.pop('file_type', None)
+        r2d2_config['obs_types'] = [ob['obs space']['name']]
+        # get bias files if needed
+        if 'obs bias' in ob.keys():
+            r2d2_config['type'] = 'bc'
+            r2d2_config['provider'] = 'gsi'
+            r2d2_config['start'] = config['prev_valid_time']
+            r2d2_config['end'] = r2d2_config['start']
+
+            # fetch satbias
+            r2d2_config['file_type'] = 'satbias'
+            target_file = ob['obs bias']['input file']
+            r2d2_config['target_file_fmt'] = target_file
+            r2d2_config['experiment'] = config.get('experiment', 'oper_gdas')
+            ufsda.r2d2.fetch(r2d2_config)
+
+            # fetch satbias_cov
```

Comment:
Same comment as above, I think `satbias_cov` is non-standard R2D2, which is fine, but we need to be aware of that.

Reply:
Comment added for `satbias_cov`.  Done at [564a4cd](https://github.com/NOAA-EMC/GDASApp/pull/55/commits/564a4cdd53e312925018e274b27e70d1035efece)